src/runtime-rs/crates/hypervisor/src/device/mod.rs

```diff
@@ -56,6 +57,7 @@ impl DeviceType {
                 Ok(DeviceType::ShareFsMount(share_fs_mount.clone()))
             }
             DeviceType::HybridVsock(hvsock) => Ok(DeviceType::HybridVsock(hvsock.clone())),
+            DeviceType::Vsock(_) => Err(anyhow!("DeviceType::Vsock cannot be cloned")),
```

Comment:
IMO, "DeviceType::Vsock cannot be cloned" maybe not accurate, what's about "unsupported Vsock device type"

Reply:
Good point, thanks. I haven't really looked into what it would take to make `Vsock` cloneable but it might well be possible to add cloning support to `Vsock` after all even if it perhaps wouldn't be trivial.